convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -1320,21 +1320,115 @@ def test_fix_default_kernel_with_no_incorrect_kernel(self):
+            "kernel-core-0:4.6~pre16262021g84ef6bd9-3.fc35",
             0,
+            id="NEVR",
         ),
-        ("2:8.2.3568-1.fc35", "2:8.2.3568-1.fc35", 0),
+        pytest.param("kernel-core-2:8.2.3568-1.fc35", "kernel-core-2:8.2.3568-1.fc35", 0, id="NEVR"),
+        pytest.param(
+            "1:NetworkManager-1.18.8-2.0.1.el7_9.aarch64", "1:NetworkManager-1.18.8-1.0.1.el7_9.aarch64", 1, id="ENVRA"
+        ),
+        pytest.param("1:NetworkManager-1.18.8-2.0.1.el7_9", "1:NetworkManager-1.18.8-3.0.1.el7_9", -1, id="ENVR"),
+        pytest.param("NetworkManager-1.18.8-2.0.1.el7_9", "1:NetworkManager-2.18.8-3.0.1.el7_9", -1, id="NVR&ENVR"),
+        pytest.param("2:NetworkManager-1.18.8-2.0.1.el7_9", "0:NetworkManager-1.18.8-3.0.1.el7_9", 1, id="ENVR"),
     ),
 )
 def test_compare_package_versions(version1, version2, expected):
     assert pkghandler.compare_package_versions(version1, version2) == expected
 
 
+@pytest.mark.parametrize(
+    ("version1", "version2", "exception_message"),
+    (
+        ("kernel-core-0:390-287.fc36", "kernel-0:390-287.fc36", "kernel-core does not match kernel"),
+        ("kernel-core-0:390-287.fc36.aarch64", "kernel-core-0:391-287.fc36.i86", "aarch64 does not match i86"),
+    ),
+)
+def test_compare_package_versions_warnings(version1, version2, exception_message):
+    with pytest.raises(ValueError, match=exception_message):
+        pkghandler.compare_package_versions(version1, version2)
+
+
+# format: name, epoch, version, release, arch
+PACKAGE_FORMATS = (
+    pytest.param(
+        "kernel-core-0:4.18.0-240.10.1.el8_3.i86", ["kernel-core", "0", "4.18.0", "240.10.1.el8_3", "i86"], id="NEVRA"
+    ),
+    pytest.param(
+        "kernel-core-0:4.18.0-240.10.1.el8_3", ["kernel-core", "0", "4.18.0", "240.10.1.el8_3", None], id="NEVR"
+    ),
+    pytest.param(
+        "1:NetworkManager-1.18.8-2.0.1.el7_9.aarch64",
+        ["NetworkManager", "1", "1.18.8", "2.0.1.el7_9", "aarch64"],
+        id="ENVRA",
+    ),
+    pytest.param(
+        "1:NetworkManager-1.18.8-2.0.1.el7_9", ["NetworkManager", "1", "1.18.8", "2.0.1.el7_9", None], id="ENVR"
+    ),
+    pytest.param(
+        "NetworkManager-1.18.8-2.0.1.el7_9.aarch64",
+        ["NetworkManager", None, "1.18.8", "2.0.1.el7_9", "aarch64"],
+        id="NVRA",
+    ),
+    pytest.param(
+        "NetworkManager-1.18.8-2.0.1.el7_9", ["NetworkManager", None, "1.18.8", "2.0.1.el7_9", None], id="NVR"
+    ),
+)
+
+
+@pytest.mark.skipif(pkgmanager.TYPE == "yum", reason="cannot test dnf backend if dnf is not present")
+@pytest.mark.parametrize(
+    ("package", "expected"),
+    (PACKAGE_FORMATS),
+)
+def test_parse_pkg_string_dnf(package, expected, monkeypatch):
+    assert pkghandler.parse_pkg_string(package) == expected
+    parse_pkg_with_dnf_mock = mock.Mock()
+    monkeypatch.setattr(pkghandler, "_parse_pkg_with_dnf", value=parse_pkg_with_dnf_mock)
+    pkghandler.parse_pkg_string(package)
+    parse_pkg_with_dnf_mock.assert_called_once()
```

Comment:
Not sure if I understood this type of test here... We are calling `parse_pkg_string` and then asserting it with expected, then, we are mocking the `_parse_pkg_with_dnf` and asserting it was called? That doesn't make too much sense to me. 

Probably we could re-write it to something like: 

```suggestion
    assert pkghandler._parse_pkg_with_dnf(package) == expected
```

The function seems simple enough that we could just call it, especially because we have a conditional skip for this test depending on the backend.